### ISSUE TITLE
Remix should not rely on Response.headers.raw

### ISSUE THREAD

**Issue by benmerckx** [remix-run/remix#4354]
### What version of Remix are you using?

1.7.2

### Steps to Reproduce

Any `Response` object that is created through other means than `@remix-run/node` (non node-fetch polyfills/undici/node builtins) fails here because Remix relies on the non standard `Reponse.headers.raw`:

https://github.com/remix-run/remix/blob/e260b2ba153271fed2e2c569abdfecdccf241625/packages/remix-express/server.ts#L125-L129

### Expected Behavior

Remix should use the standard `Response.headers` interface to maximize compatibility, eg:

````ts
  for (let [key, value] of nodeResponse.headers.entries()) {
    res.append(key, value);
  }
````

### Actual Behavior

Handing off a `Response` to Remix results in an error:

````ts
TypeError: nodeResponse.headers.raw is not a function
    at sendRemixResponse (C:\projects\alinea\examples\framework-remix\node_modules\@remix-run\express\dist\server.js:89:65)
    at C:\projects\alinea\examples\framework-remix\node_modules\@remix-run\express\dist\server.js:40:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
````

**Comment by kiliman** [comment]
I think they did it at one point due to an error in how duplicate headers were being handled in `node-fetch`. Headers like `set-cookie` were combined into a single header with comma-delimited values which was incorrect.

Now that Remix uses its own fetch implementation,  I think we should go back to `response.headers.entries()`

**Comment by MichaelDeBoey** [comment]
Closed by #7150
